source/violation.c

```diff
@@ -399,3 +407,189 @@ candl_violation_p candl_violation(osl_scop_p orig_scop,
   return violation;
 }
 
+static candl_violation_p violation_helper(osl_dependence_p orig_dependence,
+    osl_relation_p context, osl_statement_p source, osl_statement_p target,
+    int nb_parameters, candl_options_p options, PipOptions * pip_options) {
```

Comment:
We should have a documentation policy somehow.  Maintaining documentation for all utility functions is hard and I am not convinced it is useful, look at all the documentation in Candl saying 'CloogViolation', nobody reads it.

I am used to having Doxygen documentation only for publicly exposed functions while keeping short non-Doxygen comments for non-obvious internal functions, especially static.  Say `something_free() // frees something` is a harmful comment because it makes you think there is useful comment and spend time on reading it while there is no extra information.
My point is that the use of candl as a library limits to `candl_dependence` and `candl_violation` calls with some eventual processing of the resulting structures.  Since it is hard to make internal things private in C, as a library user I assume that something non-documented is an internal part of the library and I must read its code before using it or find a documented way to do so.

While the documentation policy merits a separate discussion, I will add some more briefs to the functions that are meant to be publicly usable.